src/OTSession.js

```diff
@@ -1,26 +1,34 @@
 import React, { Component, Children, cloneElement } from 'react';
-import { View } from 'react-native';
+import { View, Platform, ViewPropTypes } from 'react-native';
 import PropTypes from 'prop-types';
-import { createSession, disconnectSession, setNativeEvents, OT } from './OT';
+import { createSession, disconnectSession, setNativeEvents, nativeEvents, OT } from './OT';
 import { sanitizeSessionEvents, sanitizeSignalData } from './helpers/OTSessionHelper';
 import { logOT } from './helpers/OTHelper';
 import { handleError } from './OTError';
 
 export default class OTSession extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      isConnected: false,
-      sessionInfo: null,
-    };
+  componentEvents = {
+    streamDestroyed: Platform.OS === 'android' ? 'session:onStreamDropped' : 'session:streamDestroyed',
+    streamCreated: Platform.OS === 'android' ? 'session:onStreamReceived' : 'session:streamCreated',
+  }
+
+  state = {
+    streams: [],
+    isConnected: false,
+    sessionInfo: null,
   }
 
   componentWillMount() {
     const sessionEvents = sanitizeSessionEvents(this.props.eventHandlers);
     setNativeEvents(sessionEvents);
+
```

Comment:
we should be using `OT.setJSComponentEvents` b/c component event listeners should be separate from the event listeners that the developer will set

Reply:
What is the benefit of creating a streams state and passing it into the child components if we replicate the event listeners in the `OTSubscriber` component as well.

Reply:
Ok gotcha will take a look 

about `OTSubscriber` <= events here doesnt triggering when im trying to connect to already existing stream 

So i'm always see `this.state.streams = []` and it's working only when somebody will reconnect ( but i will see only this guy not all )

Reply:
@Necromant1k good point, let me investigate further and get back to you :)

Reply:
@msach22 i'm not sure my solution elegant but if u could fix this and do much better i'm happy with this :)

Reply:
@Necromant1k  So I ran the same code on my end and it doesn't solve the issue either. I think we may have to create a custom queue system in the native layer for both iOS and Android that can hold the events until the component event listeners are set. I have started working on that so I should have a PR on it very soon :)